deepfence_frontend/apps/dashboard/src/features/vulnerabilities/pages/RuntimeBom.tsx

```diff
@@ -18,132 +14,34 @@ import {
 import { IconMapForNodeType } from '@/features/onboard/components/IconMapForNodeType';
 import { SbomModal } from '@/features/vulnerabilities/api/sbomApiLoader';
-import { apiWrapper } from '@/utils/api';
-import { typedDefer, TypedDeferredData } from '@/utils/router';
-import { DFAwait } from '@/utils/suspense';
+import { queries } from '@/queries';
 import { getOrderFromSearchParams, useSortingState } from '@/utils/table';
 
-const PAGE_SIZE = 15;
+const PAGE_SIZE = 10;
 
-async function getScans(searchParams: URLSearchParams): Promise<{
-  scans: ModelScanInfo[];
-  currentPage: number;
-  totalRows: number;
-}> {
-  let page = toNumber(searchParams.get('page') ?? '0');
-  page = isFinite(page) && !isNaN(page) && page > 0 ? page : 0;
-  const requestFilters: SearchSearchScanReq = {
-    node_filters: {
-      filters: {
-        contains_filter: {
-          filter_in: {},
-        },
-        match_filter: { filter_in: {} },
-        order_filter: {
-          order_fields: [],
-        },
-        compare_filter: null,
-      },
-      in_field_filter: null,
-      window: {
-        offset: 0,
-        size: 0,
-      },
-    },
-    scan_filters: {
-      filters: {
-        contains_filter: {
-          filter_in: {
-            status: ['COMPLETE'],
-          },
-        },
-        match_filter: {
-          filter_in: {},
-        },
-        order_filter: {
-          order_fields: [
-            {
-              field_name: 'updated_at',
-              descending: true,
-            },
-          ],
-        },
-        compare_filter: null,
-      },
-      in_field_filter: null,
-      window: {
-        offset: 0,
-        size: 0,
-      },
-    },
-    window: {
-      offset: page * PAGE_SIZE,
-      size: PAGE_SIZE,
-    },
-  };
-  const order = getOrderFromSearchParams(searchParams);
-  if (order) {
-    requestFilters.node_filters.filters.order_filter.order_fields = [
-      {
-        field_name: order.sortBy,
-        descending: order.descending,
-      },
-    ];
-  }
-  const searchVulnerabilityScanApi = apiWrapper({
-    fn: getSearchApiClient().searchVulnerabilityScan,
-  });
-  const searchVulnerabilityScanResponse = await searchVulnerabilityScanApi({
-    searchSearchScanReq: requestFilters,
-  });
-  if (!searchVulnerabilityScanResponse.ok) {
-    throw new Error('unknown response');
-  }
-
-  const searchVulnerabilityScanCountApi = apiWrapper({
-    fn: getSearchApiClient().searchVulnerabilityScanCount,
-  });
-  const searchVulnerabilityScanCountResponse = await searchVulnerabilityScanCountApi({
-    searchSearchScanReq: {
-      ...requestFilters,
-      window: {
-        ...requestFilters.window,
-        size: 10 * requestFilters.window.size,
-      },
-    },
+const RuntimeBom = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const { data } = useSuspenseQuery({
+    ...queries.vulnerability.scanList({
+      pageSize: PAGE_SIZE,
```

Comment:
This is not correct, page size is dynamic. See other pages like scan list or scan results list.